File: Readability/readability.c
#include "cs50.h"
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
// functions
int count_letters(string txt);
int count_words(string txt);
int count_sentences(string txt);
int main(void)
{
    // promt the user for some text
    string text = get_string("Text: ");
    // count letter, words and sentences
    // count lettes
    int n_letters = count_letters(text);
    // printf("letters: %d\n",n_letters);
    // count words
    int n_words = count_words(text);
    // printf("words: %d\n", n_words);
    //   count sentences
    int n_sentences = count_sentences(text);
    // printf("sentences: %d\n", n_sentences);
    //   compute the Coleam Liau Index
    float L = ((float) n_letters / (float) n_words) * 100; // Type casting is applied here (float)
    // printf("L: %f\n",L);
    float S = ((float) n_sentences / (float) n_words) * 100;
    // printf("S: %f\n", S);
    float index = (0.0588 * L - 0.296 * S - 15.8);
    // printf("index: %f\n", index);
    //   print the grade level
    int grade = round(index);
    if (index < 16 && index > 1)
    {
        printf("Grade %d\n", grade);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
}
int count_letters(string txt)
{
    int n = strlen(txt);
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (!isalpha(txt[i])) // check wheter if a character is alphabetic
        {
            count++;
        }
    }
    int num = n - count;
    // printf("Number of letters: %i\n", n - count);
    return num;
}
int count_words(string txt)
{
    int n = strlen(txt);
    int count = 0;
    for (int i = 0; i < n + 1; i++) // here I add +1 to go one place far form the len of the string
    {
        if (txt[i] == ' ' || txt[i] == '\0') // compare if there is an space or the text has finished
        {
            count++;
        }
    }
    return count;
}
int count_sentences(string txt)
{
    int n = strlen(txt);
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (txt[i] == '?' || txt[i] == '!' || txt[i] == '.') // These simbols (?, !, .) indicate the end of a sentences
        {
            count++;
        }
    }
    return count;
}

File: sentimental-readability/readability.py
# get text from user
def get_text(prompt):
    text = input(prompt)
    return text

# compute number of letters


def count_letters(text):
    number_letters = 0
    spaces = 0
    for i in text:
        if not i.isalpha():
            spaces += 1
    number_letters = len(text) - spaces
    return number_letters

# compute number of words


def count_words(text):
    number_words = 0
    for i in text:
        if i == " ":
            number_words += 1
    return number_words + 1

# compute number of sentences


def count_sentences(text):
    number_sentences = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            number_sentences += 1
    return number_sentences

# Apply Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8.


def grade(text):
    L = (count_letters(text) / count_words(text)) * 100
    S = (count_sentences(text) / count_words(text)) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)
    if 1 < grade < 16:
        return f"Grade {grade}"
    if grade >= 16:
        return f"Grade 16+"
    if grade < 1:
        return f"Before Grade 1"


def main():
    txt = get_text("Enter the text: ")
    '''
    letters = count_letters(txt)
    words = count_words(txt)
    sentences = count_sentences(txt)
    print(f"Letters: {letters}, Words: {words}, Sentences: {sentences}")
    '''
    result = grade(txt)
    print(result)

    
if __name__ == "__main__":
    main()
